src/lexer/lex.c

```diff
@@ -83,6 +83,12 @@ int is_valid_numeric_or_id_char(char c) {
     return isalnum(c) || (c == '_') || (c == '.');
 }
 
+int is_valid_string_char(char c, int* type) {
```

Comment:
What is `type` used for here? It doesn't seem to be used in the logic of the function and isn't accessible outside of it.